### DISCUSSION TITLE
How to manage authentication for container registries using PACK?

### DISCUSSION THREAD

**Discussion by jigth98** [buildpacks/community#203] [Q&A] [answered]
Greetings, I would like to know what options are currently available to authenticate with registries in general when using the **pack CLI tool**.

I already did some test and was able to push the Java's Hello World example to a **DockerHub** public and a private repo. I think it was due to having the **~/.docker/config.json** file properly configured after using the **docker login** command.

That option could be good enough for a local use case but are there any other options available for authentication using the **pack CLI**?

I would like to know if there's some way of providing a TOKEN authentication or "one TIME authentication" where some user can provide enough information to push the image to their **container registry** but without having to provide the real "username" and "pasword".

Else, a temporary configuration file with authentication details or a file that takes info from the ENV variables could be useful as well, as long as it's a scalable way that allows an application to process and upload multiple container images simultaneously using the **pack CLI** (i.e: running the CLI tool from more than one OS process at a time to generate multiple container images from source code simultaneously - I'm using a Unix based Distro).

Is there anything like that? Do you know where could I find more info about that topic?

I already searched in both the documentation and some Issues but didn't find the full answer I'm looking for, I know it's possible to push images to different container registries but what I'm not fully aware of is the possible authentication methods (preferably one that does not have risk to disclose personal account details of an application's user).

If that is not possible with the aforementioned CLI but it is with another tool you recommend I would like any tips on that.

Thank you in advance!

**Comment by dfreilich** [comment, marked as answer]
Pack doesn't directly manage authentication, it uses local docker authentication for that. We didn't want to introduce our own authentication into the mix, and somehow expose things - instead, if you want to use a TOKEN or one-time authentication, you'd have to see how your container registry provides that for normal docker commands, and just use `docker login` to set the authentication.

**Reply by jigth98** [reply]
Ok, I understand. Thank you very much for the prompt answer. Have a good day!
